src/bootstrap/bin/rustc.rs

```diff
@@ -70,6 +70,29 @@ fn main() {
                 cmd.arg("-Ztime-passes");
             }
         }
+
+        if let Some(profile) = env::var_os("RUSTC_TUNE_COMPILER_CODEGEN_UNITS") {
+            let cgus = match crate_name {
+                // This crate is large and at the tail of the compiler crates, give it extra CGUs.
+                //"rustc_query_impl" => Some(96),
+                // This compiles after all other crates so give it default CGUs to speed it up.
+                "rustc_driver" => None,
+                _ => {
+                    if profile == "fast" {
+                        Some(1)
+                    } else {
+                        if crate_name.starts_with("rustc_") {
+                            None
+                        } else {
+                            // Compile crates.io crates with a single CGU for faster compile times
+                            Some(1)
```

Comment:
I don't understand - doesn't decreasing the CGUs *slow down* compilation because we aren't running multiple LLVM threads in parallel? Or do you mean this makes the compiler faster at runtime in exchange for slower bootstrap times? In that case "fast" seems like a misleading name ...

Reply:
All the crates are still compiled in parallel even if a single crate is limited to 1 thread.

Reply:
Right, but rustc_driver still can't start building until all other crates have finished, and this is still pushing up the overall build time.

I'm not saying it's a bad change, for CI it makes sense the same way that PGO makes sense, I'd just like to find a different name. Maybe "reduce-codegen-units" or something like that?

Reply:
It reduces build time with 7 CPU cores, it may increase it for high core count CPUs, so I agree the name is not ideal. Maybe `codegen-units-reduce`?

Reply:
👍

Reply:
This will be most useful in combination with LTO, rarely by itself right ? So maybe it could be one of the LTO options, since that's already an enum that doesn't exactly match `-C lto`'s values (e.g. "thin-local"). Something like `rust.lto = "thin-1cgu"` ?

Reply:
Not quite. This option is intended to reduce the compilation time of rustc. LTO increases compile time to give better runtime performance and is better paired with the existing `codegen-units=1` so all crates have a single CGU.